src/service/ComposeDocument.ts

```diff
@@ -341,7 +354,13 @@ const ValueRegex = /^(?<indent> *)(?<value>\S+)$/im;
 // A regex for matching a whitespace-only line
 const WhitespaceRegex = /^(?<indent> *)$/im;
 
+// A regex for matching any line with a comment (or if the whole line is a comment)
+// TODO: This will match a 'string with a #', which is *not* a comment, but currently this will not affect any end scenarios
+// TODO: The closest I've gotten is /(['"]).*(?<commentSep>#)(?:(?!\1).)*$/im

```

Comment:
Yes, that is a tough one, esp. considering the string literal may contain escaped single/double quotes. Also do not forget about interpolated strings 😬 Maybe there is some "good enough" prior art (regex) for that and we can get away with a test that checks whether the pound sign is within string literal boundaries.

Reply:
Would it help if we had separate expressions for single-quoted and double-quoted string liteals?

Reply:
No, since we only need to assert that the end (closing quote) is the same as the beginning (opening quote), which we can do with a backreference

Reply:
Is it worth manually parsing it if it is one line? 🤔

Reply:
i.e. with a stack that adds/pops quotes/etc.?

Reply:
I think another question worth asking though is--does it matter or is this good enough? I believe the only place you can put a `#` and it _not_ be a comment is in a quoted scalar value; the only scalar values we currently care about for any purpose are the values in ports and volumes (for completions), but AFAIK neither of those could have a `#` and remain valid.